src/ui/FacetValueSuggestions/FacetValueSuggestions.ts

```diff
@@ -142,9 +142,13 @@ export class FacetValueSuggestions extends Component {
 
   private queryStateFieldFacetId;
 
+  private suggestionHTMLRepresentation: Map<string, string>;
```

Comment:
I think having this private attribute here introduces a state that is kind of hard to debug. It also complexifies the `defaultTemplate`.

I wonder if it would be cleaner to pass down the html representation to the `FacetValueSuggestionsProvider`. It would allow you to add a `.html` attribute to the `IFacetValueSuggestionRow` interface and you could do:

```
const keyword = row.html || DomUtils.highlightElement(row.keyword, omnibox.getText(), 'coveo-omnibox-hightlight')
```

Reply:
That's what I was tempted to do in the first place, but it looks like that would involve modifying `FacetValueSuggestions.getQuerySuggestionsKeywords`, `FacetValueSuggestions.getFacetValueSuggestions`, `FacetValueSuggestions.getSuggestionsForWords`, `FacetValueSuggestionsProvider.getSuggestions`, `FacetValueSuggestionsProvider.getFieldValuesToQuery`, `IFacetValueBatchResponse` and `IFacetValueSuggestionRow`, I wasn't sure if it was a good idea to make such a change.

If you think that would be a good idea, I could definitely do it though.